src/cdomains/apron/linearTwoVarEqualityDomain.apron.ml

```diff
@@ -0,0 +1,779 @@
+    begin match next_var with
+      | None -> (* this element represents a constant offset *)
+        (var, current_var_offset, Z.(curr_offset + next_coeff))
+      | Some _ -> (* this element represents a variable with a coefficient
+                            -> it must be always the same variable, else it's not a two-variable equality*)
+        begin if Option.is_none var || next_var = var then
+            (next_var, Z.(current_var_offset + next_coeff), curr_offset)
+          else raise Not2VarExpr end
+    end in
+    let sum_coefficients summands_list_opt =
+    Option.map (List.fold_left sum_next_coefficient (None, Z.zero, Z.zero)) summands_list_opt
+    in
+    match sum_coefficients (get_coeff_vec t texp) with
+    | exception _ -> None
+    | None -> None
+    | Some (var, var_coeff, offset) ->
+    if Option.is_none var then Some (None, offset)
+    else if Z.equal var_coeff Z.one then Some (var, offset)
+    else None
+  *)
+
+
+  let get_coeff t texp = timing_wrap "coeff_vec" (get_coeff t) texp
+
+  let abstract_exists var t = match t.d with
+    | Some d -> {t with d = Some (EArray.forget_variable d (Environment.dim_of_var t.env var))}
+    | None -> t (* there are no  variables in the current environment *)
+
+  (* Copy because function is not "with" so should not mutate inputs *)
+  let assign_const t var const = match t.d with
+    | None -> t
+    | Some t_d ->
+      let d = EArray.copy t_d in d.(var) <- (None, const);  {d = Some d; env = t.env}
+
+  let subtract_const_from_var t var const =
+    match t.d with
+    | None -> t
+    | Some t_d ->
+      let d = EArray.copy t_d in
+      let subtract_const_from_var_for_single_equality index (eq_var_opt, off2) =
+        if index <> var then
+          begin match eq_var_opt with
+            | Some eq_var when eq_var = var ->
+              d.(index) <- (eq_var_opt, Z.(off2 - const))
+            | _ -> ()
+          end
+      in
+      begin if d.(var) = (Some var, Z.zero)
+      (* var is a reference variable -> it can appear on the right-hand side of an equality *)
+        then
+          EArray.iteri (subtract_const_from_var_for_single_equality) d
+        else
+          (* var never appears on the right hand side-> we only need to modify the array entry at index var *)
+          d.(var) <- Tuple2.map2 (Z.add const) d.(var)
+      end;
+      {d = Some d; env = t.env}
+
+end
+
+
+module ExpressionBounds: (SharedFunctions.ConvBounds with type t = VarManagement.t) =
+struct
+  include VarManagement
+
+  let bound_texpr t texpr = let texpr = Texpr1.to_expr texpr in
+    match get_coeff t texpr with
+    | Some (None, offset) -> Some offset, Some offset
+    | _ -> None, None
+
+
+  let bound_texpr d texpr1 =
+    let res = bound_texpr d texpr1 in
+    match res with
+    | Some min, Some max ->  if M.tracing then M.tracel "bounds" "min: %s max: %s" (IntOps.BigIntOps.to_string min) (IntOps.BigIntOps.to_string max); res
+    | _ -> res
```

Comment:
It seems this is a more concise version that does the same thing?

```suggestion
  let bound_texpr t texpr = 
    match get_coeff t (Texpr1.to_expr texpr) with
    | Some (None, offset) ->
       (if M.tracing then M.tracel "bounds" "min: %s max: %s" (IntOps.BigIntOps.to_string offset) (IntOps.BigIntOps.to_string offset);
      Some offset, Some offset)
    | _ -> None, None
```

Reply:
I think we implemented it the old way because the tracing was normally added in a second function definition. It would potentially be strange to change this style here